### DISCUSSION TITLE
Batch processing question

### DISCUSSION THREAD

**Discussion by gwpj** [ImageMagick/ImageMagick#2614] [Help]
Hello 
Greetings from South Africa
I have a folder with 2000 sequential QR Codes that are in PDF format - one PDF file per QR Code.
I would like to put them into multiple pages that contain 8 rows of 18 images at a size of 25.4mm x 25.4mm so that I can import them into CorelDraw for laser engraving.  The size of the material I am engraving on to is 600mm x 300mm, so each QR Code will be in a 30mm x 30mm square giving (18x8) 144 QR Codes per page and I will need 14 pages.
I am at a loss of how to do this and any help and advice would be greatly appreciated.
Many thanks
Cheers
Gareth

**Comment by snibgo** [comment]
Your input PDF files may record QR codes either as raster or vector. If they are raster, I suggest you extract them with pdfimages.

ImageMagick works primarily in pixels, without caring much about the density (pixels per centimetre or whatever). How many pixels are in each 25.4mm x 25.4mm (or 30x30mm) image?

What format of file do you need for the engraving? You might need SVG, or a raster format such as PNG.

If all your input rasters will fit in your memory at the same time, the task is fairly simple. But you might need a shell loop, to process each of the 14 pages.

**Reply by gwpj** [reply]
Hello snibgo
Thank you for your response.
I have attached the PDF so you can see the size.
[FH0001.pdf](https://github.com/ImageMagick/ImageMagick/files/5276053/FH0001.pdf)
It also seems to have quite a big border that needs to be cropped so that the QR Code is 25mm square in a 30mm square box without borders.

A good quality PDF will be fine for CorelDraw but I can also import PNG files.

I have a Mac with 16GB of memory.
Cheers  Gareth

**Comment by snibgo** [comment]
What version of IM do you use?

FH0001.pdf has an image that is both vector and raster. The central eagle is raster; the rest is vector. So I rasterize it with IM (via Ghostscript).
```
magick -density 300 FH0001.pdf -alpha off -threshold 50% -trim +repage x.png
```
I rasterize at 300 dpi. This is a reasonable number for printers, but I have no idea what your laser engraver needs. I suppose alpha is irrelevant, so I remove it. The image has anti-aliasing, but I suppose you need black-and-white-only, so I threshold it. "-trim +repage" removes the white border, so the black marks go all the way to the image edge. 

The result is 987x987 pixels.
![x](https://user-images.githubusercontent.com/33812368/94157589-5217a280-fe79-11ea-8150-fec42437a7e5.png)

**Reply by gwpj** [reply]
Thank you very much.
300dpi is perfect for the laser. I will do an engraving test with this file that you kindly made.
You are correct no alpha is needed and no antialiasing.
I am busy downloading and installing the latest version of IM on my Mac.
I am new to this so I am learning along the way.
Many thanks for your kind assistance.
